js/views/homeVw.js

```diff
@@ -186,14 +187,31 @@ module.exports = Backbone.View.extend({
     item.isBlocked = blocked.indexOf(item.guid) !== -1;
 
     var newItem = function(){
-      var newItemModel = new itemShortModel(item);
-      var itemShort = new itemShortView({model: newItemModel});
+      var newItemModel,
+          itemShort;
+
+      if (item.isBlocked) {
+        self.setListingsBlockedCount(self.getListingsBlockedCount() + 1);
+        return;
+      }
+
+      newItemModel = new itemShortModel(item);
+      itemShort = new itemShortView({model: newItemModel});
+
+      self.listenTo(newItemModel, 'change:isBlocked', function(model, isBlocked, options) {
+        if (isBlocked) {
+          self.removeItemView(itemShort);
+          self.setListingsBlockedCount(self.getListingsBlockedCount() + 1);
+        }
+      });
 
       self.listenTo(itemShort, 'blockUserClick', self.blockUserClick);
       self.listenTo(itemShort, 'unblockUserClick', self.unblockUserClick);
 
+      item.isBlocked && itemShort.$el.hide().addClass('blocked-user');
       self.$el.find('.js-products .js-loadingSpinner').before(itemShort.el);
-      self.subViews.push(itemShort);
+      self.registerChild(itemShort);
```

Comment:
The base view provides a method called registerChild. Any time you create a child view, call registerChild and pass in the child view you created. The main thing this will gain you is that when your parent view is removed, any registered children will be removed.